sequence_task_3/student/src/impl.cpp

```diff
@@ -2,18 +2,22 @@
 #include <algorithm>
 #include <string>
 #include <numeric>
-#include <iostream>
+#include <sstream>
 
 std::string reverseWords(const std::string& input) {
-    std::deque<std::string> strDeq;
+    std::deque<std::string> dequeResultString;
     std::string world;
-    std::stringstream myString(input);
+    std::stringstream inputString(input);
 
-    while (myString >> world) {
-        strDeq.push_front(world);
-        strDeq.push_front(" ");
-    }
+    for_each(std::istream_iterator<std::string>(inputString), std::istream_iterator<std::string>(),
+        [&dequeResultString](const std::string& word) { dequeResultString.push_front(word);
+                                                        dequeResultString.push_front(" ");});
```

Comment:
please use ostream iterators for formatting, and solution will be simpler and clearer.
But cool, this way is something new :)